frontend/apps/filemanager/filemanagerbookmark.lua

```diff
@@ -0,0 +1,192 @@
+local ButtonDialog = require("ui/widget/buttondialog")
+local InfoMessage = require("ui/widget/infomessage")
+local InputDialog = require("ui/widget/inputdialog")
+local Menu = require("ui/widget/menu")
+local Screen = require("device").screen
+local UIManager = require("ui/uimanager")
+local util = require("ffi/util")
+local _ = require("gettext")
+local T = require("ffi/util").template
+
+local FileManagerBookmark = Menu:extend{
+    width = Screen:getWidth(),
+    height = Screen:getHeight(),
+    no_title = false,
+    parent = nil,
+    has_close_button = true,
+    is_popout = false,
+    is_borderless = true,
+}
+
+function FileManagerBookmark:init()
+    self.item_table = self:genItemTableFromRoot()
+    Menu.init(self)
+end
+
+function FileManagerBookmark:genItemTableFromRoot()
+    local item_table = {}
+    local favorite_folder = G_reader_settings:readSetting("fm_bookmark") or {}
+    table.insert(item_table, {
+        text = _("Add new favorite folder"),
+        callback = function()
+            self:addNewFolder()
+        end,
+    })
+    for _, item in ipairs(favorite_folder) do
+        table.insert(item_table, {
+            text = string.format("%s (%s)", item.text, item.folder),
+            folder = item.folder,
+            friendly_name = item.text,
+            deletable = true,
+            editable = true,
+            callback = function()
+                UIManager:close(self)
+                self.goFolder(item.folder)
+            end,
+        })
+    end
+    return item_table
+end
+
+function FileManagerBookmark:addNewFolder()
+    local PathChooser = require("ui/widget/pathchooser")
+    local path_chooser = PathChooser:new{
+        select_directory = true,
+        select_file = false,
+        path = self.curr_path,
+        onConfirm = function(path)
+            local add_folder_input
+            local friendly_name = util.basename(path) or _("my folder")
+            add_folder_input = InputDialog:new{
+                title = self.title,
+                input = friendly_name,
+                input_type = "text",
+                description = T(_("Please enter friendly name for your selected folder:\n%1"), path),
```

Comment:
`Enter (display) title/name for selected folder`

Or just:
`Title for selected folder:`